eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -197,39 +197,97 @@ object Transactions {
 
   def decodeTxNumber(sequence: Long, locktime: Long): Long = ((sequence & 0xffffffL) << 24) + (locktime & 0xffffffL)
 
-  def makeCommitTx(commitTxInput: InputInfo, commitTxNumber: Long, localPaymentBasePoint: PublicKey, remotePaymentBasePoint: PublicKey, localIsFunder: Boolean, localDustLimit: Satoshi, localRevocationPubkey: PublicKey, toLocalDelay: CltvExpiryDelta, localDelayedPaymentPubkey: PublicKey, remotePaymentPubkey: PublicKey, localHtlcPubkey: PublicKey, remoteHtlcPubkey: PublicKey, spec: CommitmentSpec): CommitTx = {
+
+  /**
+   * Represent a link between a commitment spec item (to-local, to-remote, htlc) and the actual output in the commit tx
+   * @param output transaction output
+   * @param redeemScript redeeem script that matches this output (most of them are p2wsh)
+   * @param link commitment spec item this output is built from
+   */
+  case class CommitmentOutputLink(output: TxOut, redeemScript: Seq[ScriptElt], link: CommitmentSpecLink)
```

Comment:
Same here, I think that the field `link` is really confusing.
The description of the param suggests a better name, maybe `specItem` or `item`?
I'm not a big fan of these suggestions, we need to bikeshed a bit more. This is a very critical part of the codebase, so I think we really need to spend time making it as clear as possible.

Reply:
I came up with the name `link` because this holds together the  output of the transaction with the output in our `CommitmentSpec`. Since this is very critical it's important that everyone undestands it and we should definitely bikeshed until we agree on the naming.

Reply:
Okay i have some proposals for the naming:
- Keep class name `CommitmentOutputLink` but field name `link` -> `specItem`
- Keep class name `CommitmentOutputLink` but field name `link` -> `commitmentSpecItem`
- Keep class name `CommitmentOutputLink` but field name `link` -> `specItem` and `output` -> `txOut`
- Rename `CommitmentOutputLink` to `CommitmentSpecOutput`
- Rename `CommitmentOutputLink` to `CommitmentSpecOutput` and field `link` -> `specItem`

Reply:
If we rename `CommitmentSpecLink` to `CommitmentOutput` (see my other comment), my preference would be to keep this class named `CommitmentOutputLink` and rename the field `link` -> `specOutput: CommitmentOutput` or `specItem: CommitmentOutput`

Reply:
`CommitmentSpecLink` is already gone in dddac35 :). I agree with your suggestion to rename the `link` field to `specItem: CommitmentOutput`, let's see if there are other suggestions/ACK so we can move forward.